part-2.py

```diff
@@ -5,12 +5,43 @@
 
 # search
 
-
+def search(array, query):
+    for item in array:
+        if isinstance(item, list):
+            return search(array, query)
+        if query in array:
+            return True
+    else:
+        return False
 
 # is_palindrome
 
-
+def is_palindrome(text):
+    
+    if len(text) <= 1:
+        return True
+        
+    elif text[0] != text[-1]:
+        return False
+        
+    else:
+        return is_palindrome(text[1:-1])
 
 # digit_match
 
+def digit_match(num1, num2):
```

Comment:
Nice work on this one.  the other way is to use `//` and `%` to pluck off and remove the smallest digit.